spec/v2/providers/alerts/alerts.spec.ts

```diff
@@ -0,0 +1,211 @@
+import { expect } from 'chai';
+import { CloudEvent, CloudFunction } from '../../../../src/v2/core';
+import * as options from '../../../../src/v2/options';
+import * as alerts from '../../../../src/v2/providers/alerts';
+import { FULL_ENDPOINT, FULL_OPTIONS } from '../helpers';
+
+const ALERT_TYPE = 'new-alert-type';
+const APPID = '123456789';
+
+function getMockFunction(): CloudFunction<alerts.FirebaseAlertData<String>> {
+  const func = (raw: CloudEvent<unknown>) => 42;
+  func.run = (event: CloudEvent<alerts.FirebaseAlertData<String>>) => 42;
+  func.__endpoint = {};
+  return func;
+}
+
+describe('alerts', () => {
+  describe('onAlertPublished', () => {
+    it('should create the function without opts', () => {
+      const result = alerts.onAlertPublished(ALERT_TYPE, () => 42);
+
+      expect(result.__endpoint).to.deep.equal({
+        platform: 'gcfv2',
+        labels: {},
+        eventTrigger: {
+          eventType: alerts.eventType,
+          eventFilters: {
+            alertType: ALERT_TYPE,
+          },
+          retry: false,
+        },
+      });
+    });
+
+    it('should create the function with opts', () => {
+      const result = alerts.onAlertPublished(
+        {
+          ...FULL_OPTIONS,
+          alertType: ALERT_TYPE,
+          appId: APPID,
+        },
+        () => 42
+      );
+
+      expect(result.__endpoint).to.deep.equal({
+        ...FULL_ENDPOINT,
+        eventTrigger: {
+          eventType: alerts.eventType,
+          eventFilters: {
+            alertType: ALERT_TYPE,
+            appId: APPID,
+          },
+          retry: false,
+        },
+      });
+    });
+
+    it('should have a .run method', () => {
+      const func = alerts.onAlertPublished(ALERT_TYPE, (event) => event);
+
+      const res = func.run('input' as any);
+
+      expect(res).to.equal('input');
+    });
+  });
+
+  describe('getEndpointAnnotation', () => {
+    beforeEach(() => {
+      process.env.GCLOUD_PROJECT = 'aProject';
+    });
+
+    afterEach(() => {
+      options.setGlobalOptions({});
+      delete process.env.GCLOUD_PROJECT;
+    });
+
+    it('should define the endpoint without appId and opts', () => {
+      const func = getMockFunction();
```

Comment:
Here and onward: Now that getEndpointAnnotation is a purefunction, it looks like we don't need any of the `getMockFunction` artifacts, e.g.:

```js
expect(alerts.getEndpointAnnotation(...)).to.deep.equal.to(...)
```

should suffice?

Reply:
Fixed